### DISCUSSION TITLE
Need help with Paramiko Error

### DISCUSSION THREAD

**Discussion by ninazemnik** [StackStorm/st2#5835] [Q&A]
Hello community, has anyone had this error. Im trying to run a remote action and i'm getting the following error: Paramiko error: Incompatible ssh server (no acceptable ciphers) Anyone knows how to resolve it?

**Comment by nzlosh** [comment]
What are the supported ciphers offered by the server StackStorm is attempting to connect to?

**Comment by philipphomberger** [comment]
I found that Python Code to see what chiper paramiko supportet:

import paramiko
import socket

opts = paramiko.transport.Transport(socket.socket()).get_security_options()
print(opts.ciphers)
print(opts.kex)

That are the response: 
ciphers:
('aes128-ctr', 'aes192-ctr', 'aes256-ctr', 'aes128-cbc', 'aes192-cbc', 'aes256-cbc', '3des-cbc')
Kex: 
('curve25519-sha256@libssh.org', 'ecdh-sha2-nistp256', 'ecdh-sha2-nistp384', 'ecdh-sha2-nistp521', 'diffie-hellman-group16-sha512', 'diffie-hellman-group-exchange-sha256', 'diffie-hellman-group14-sha256', 'diffie-hellman-group-exchange-sha1', 'diffie-hellman-group14-sha1', 'diffie-hellman-group1-sha1')
